File: Day 16/OddEvenLinkedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        ListNode oddHead = new ListNode(-1);
        ListNode evenHead = new ListNode(-1);
        ListNode oddCurr = oddHead;
        ListNode evenCurr = evenHead;

        while(head != null){
            oddCurr.next = head;
            oddCurr = oddCurr.next;

            head = head.next;

            if(head != null){
                evenCurr.next = head;
                evenCurr = evenCurr.next;
                head = head.next;
            }
        }
        evenCurr.next = null;

        oddCurr.next = evenHead.next;

        return oddHead.next;
        
    }
}

File: Day 16/OddEvenLinkedList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        ListNode *oddHead = new ListNode(-1);
        ListNode *evenHead = new ListNode(-1);
         ListNode *oddCurr = oddHead;
        ListNode *evenCurr = evenHead;

        while(head != NULL){
            oddCurr->next = head;
            oddCurr = oddCurr->next;

            head = head->next;

            if(head != NULL){
                evenCurr->next = head;
                evenCurr = evenCurr->next;
                head = head->next;
            }
        }
        evenCurr->next = NULL;

        oddCurr->next = evenHead->next;
        return oddHead->next;
    }
};